text/0042-partial-fill-rules.md

```diff
@@ -0,0 +1,296 @@
+* The same input can be used to "back" multiple SCI quotes at once, in which case if any one of them is matched, the others instantly become impossible to match. This is somewhat different from how limit orders work in an exchange -- each limit order must be backed by distinct liquidity. This reflects the idea that SCI's are more in analogy with a signed quote than with a limit order as such. This is potentially confusing to e.g. a trader who looks at a list of available SCI quotes and infers that there is a certain amount of liquidity depth, reasoning as they would in a traditional exchange. But if the same inputs are used to sign multiple quotes e.g. for different currencies, it may be impossible that all of these quotes could actually match, and the "real" liquidity may be considerably less. This can be mitigated by making sure that the software they are using communicates this idea clearly.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+An alternative design which we explored first attempted to use homomorphic properties of Pedersen commitments to verify correct partial fills without revealing them:
+
+* Amount shared secret is not needed.
+* An "Input rule verification data" field is added to TxIn, and contains the fill fraction, expressed as a u64 `numerator` and a u64 `denominator`.
+* To verify equations like `real_output >= num / denom * fractional_output`, we clear denominators and convert to a form `X >= 0`.
+  `denominator * real_output - num * fractional_output >= 0`. Then we consider the LHS as a "partial fill verification commitment"
+  and fold this into the range proofs that already exist in the transaction. It is not too hard to see that the counter-party can always
+  ensure that this number is in the range `[0, u64::MAX)` for a valid transaction.
+* To verify the fractional change output, we need to verify an equation `(fraction_change - real_change) >= num / denom * real_change`.
+  This could similarly be verified without revealing commitments by clearing denominators and converting to the form `X >= 0`, then using
+  range proofs to verify the equation.
+
+The problem with this design is that, in MobileCoin, as with all RingCT-based cryptocurrencies, the amount of a TxOut actually appears in two places -- a Pedersen commitment, and a masked_value. The masked_value is a more conventional, symmetric-key encryption of the TxOut value, while the Pedersen commitment is an elliptic curve point which commits to the value, but cannot itself be directly "decrypted" efficiently by someone who doesn't already know the value.
+
+All recipients determine the value of their TxOut's by the following process:
+* Derive the TxOut shared secret
+* Use the TxOut shared secret to derive the value mask, token id mask, and the amount commitment blinding factor. This computation never fails with an error, but produces "conjectures" which are expected to be correct if the Recipient owns this TxOut.
```

Comment:
:wrench: couple of suggestions
1. Don't think conjectures belongs in quotes
2. Don't think recipient should be title cased
```suggestion
* Use the TxOut shared secret to derive the value mask, token id mask, and the amount commitment blinding factor. This computation never fails with an error, but produces conjectures which are expected to be correct if the recipient owns this TxOut.
```